src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -113,6 +113,28 @@ Model::CascadiaSettings CascadiaSettings::Copy() const
         settings->_globals = _globals->Copy();
         settings->_allProfiles = winrt::single_threaded_observable_vector(std::move(allProfiles));
         settings->_activeProfiles = winrt::single_threaded_observable_vector(std::move(activeProfiles));
+
+        // copy fragment extensions
+        {
+            std::vector<Model::FragmentSettings> fragmentExtensions;
+            fragmentExtensions.reserve(_fragmentExtensions.Size());
+            for (const auto& fragment : _fragmentExtensions)
+            {
+                fragmentExtensions.emplace_back(get_self<FragmentSettings>(fragment)->Copy());
+            }
+            settings->_fragmentExtensions = winrt::single_threaded_vector(std::move(fragmentExtensions));
+        }
+
+        // copy dynamic profile generators
+        {
+            std::vector<Model::FragmentSettings> dynamicProfileGenerators;
+            dynamicProfileGenerators.reserve(_dynamicProfileGeneratorExtensions.Size());
+            for (const auto& fragment : _dynamicProfileGeneratorExtensions)
+            {
+                dynamicProfileGenerators.emplace_back(get_self<FragmentSettings>(fragment)->Copy());
+            }
+            settings->_dynamicProfileGeneratorExtensions = winrt::single_threaded_vector(std::move(dynamicProfileGenerators));
+        }

```

Comment:
Alright, so, these are notionally immutable parts of the settings _environment_... right?

Since they're fully immutable, is there a strong need for us to deep clone them? The settings UI can't edit them after all, and they never get Saved.

Reply:
same applies below

Reply:
Good call. Changed logic, removed `FragmentSettings::Copy()`, and added a comment